### ISSUE TITLE
ERROR in @aws-sdk/util-endpoints

### ISSUE THREAD

**Issue by usmanghani5995** [aws/aws-sdk-js-v3#4067]
### Checkboxes for prior research

- [X] I've gone through [Developer Guide](https://docs.aws.amazon.com/sdk-for-javascript/v3/developer-guide) and [API reference](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest)
- [X] I've checked [AWS Forums](https://forums.aws.amazon.com) and [StackOverflow](https://stackoverflow.com/questions/tagged/aws-sdk-js).
- [X] I've searched for [previous similar issues](https://github.com/aws/aws-sdk-js-v3/issues) and didn't find any solution.

### Describe the bug

I'm using `@aws-sdk@^2.1197.0` in my `creact-react-app` project with `@aws-sdk/util-endpoints@^3.192.0` and when I run development server it throw following error.

```bash
ERROR in ../node_modules/@aws-sdk/util-endpoints/dist-es/lib/aws/partition.js 2:26-41
export 'partitions'.'find' (imported as 'partitions') was not found in './partitions.json' (possible exports: 0, 1, 2, 3, 4)
@ ../node_modules/@aws-sdk/util-endpoints/dist-es/index.js 1:0-36 1:0-36
@ ../node_modules/@aws-sdk/client-chime-sdk-messaging/dist-es/endpoint/endpointResolver.js 1:0-58 5:9-24
@ ../node_modules/@aws-sdk/client-chime-sdk-messaging/dist-es/runtimeConfig.shared.js 2:0-70 6:48-71
@ ../node_modules/@aws-sdk/client-chime-sdk-messaging/dist-es/runtimeConfig.browser.js 11:0-84 17:29-51
@ ../node_modules/@aws-sdk/client-chime-sdk-messaging/dist-es/ChimeSDKMessagingClient.js 12:0-73 15:22-40
@ ../node_modules/@aws-sdk/client-chime-sdk-messaging/dist-es/index.js 2:0-42 2:0-42
@ ../node_modules/amazon-chime-sdk-js/build/messagingsession/DefaultMessagingSession.js 35:37-83
@ ../node_modules/amazon-chime-sdk-js/build/index.js 119:34-87
@ ./src/constants/index.ts 6:0-47 25:9-23 26:8-21 27:8-21 28:9-23 29:7-19
@ ./src/meetingConfig.ts 6:0-45 10:17-31 10:50-69
@ ./src/app.tsx 15:0-44 59:15-28
@ ./src/index.tsx 10:0-24 15:40-43
```

### SDK version number

@aws-sdk/util-endpoints@^3.192.0

### Which JavaScript Runtime is this issue in?

Node.js

### Details of the browser/Node.js/ReactNative version

v14.19.0

### Reproduction Steps

n/a

### Observed Behavior

n/a

### Expected Behavior

n/a

**Comment by maoxiaoke** [comment]
Same here

**Comment by wis-dev** [comment]
Same issue here!

SDK version number
```
@aws-sdk/client-s3: 3.186.0
@aws-sdk/s3-request-presigner: 3.186.0
```
Error output on Angular@14
![image](https://user-images.githubusercontent.com/30844484/197116579-9b17933a-a8d5-433b-9962-d4e1e49b4f31.png)

![image](https://user-images.githubusercontent.com/30844484/197117269-d67d95a0-6aa4-42c3-bc3e-bcf562219a0a.png)

**Comment by vishald2509** [comment]
The issue is fixed in @aws-sdk/util-endpoints@^3.193.0. Update the package.

**Comment by plabric** [comment]
I have the same problem with amazon-chime-sdk-js@3.8.0 when adding @aws-sdk/util-endpoints@3.193.0 to try to fix the issue: https://github.com/aws/aws-sdk-js-v3/issues/4060

`Attempted import error: 'partitions'.'find' is not exported from './partitions.json' (imported as 'partitions').`

**Comment by gfinocchiaro-appdev** [comment]
"@aws-sdk/client-s3": "^3.193.0",
"@aws-sdk/node-http-handler": "^3.193.0",
"@aws-sdk/types": "^3.193.0",
"@aws-sdk/util-endpoints": "^3.193.0"

Same issue with tha above packages... here the errors:

./node_modules/@aws-sdk/util-endpoints/dist-es/lib/aws/partition.js:2:26-41 - Error: export 'partitions'.'find' (imported as 'partitions') was not found in './partitions.json' (possible exports: 0, 1, 2, 3, 4)
./node_modules/@aws-sdk/util-endpoints/dist-es/lib/aws/partition.js:4:26-36 - Error: Should not import the named export 'partitions' (imported as 'partitions') from default-exporting module (only default export is available soon)
./node_modules/@aws-sdk/util-endpoints/dist-es/lib/aws/partition.js:19:26-36 - Error: Should not import the named export 'partitions' (imported as 'partitions') from default-exporting module (only default export is available soon)

**Comment by jabedhasan21** [comment]
Yesterday I have faced the same issue.

**Error: Cannot find module '@aws-sdk/util-endpoints'**

Since this is **An internal package** so there is no point to manually add  this to `package.json` or manually install.

I spend lots of time to identify why it's happen.

But at the end when I manually install this module the issue resolved.

For temporarily fixed my problem and to stay in safe side I added this module in my package.json.

`npm install @aws-sdk/util-endpoints@3.190.0 --save`

Hope AWS SDK team will look into this issue and resolve it in a way so that we don't have to install this internal module manually.

This is my error stack trace
```
Error: Cannot find module '@aws-sdk/util-endpoints'
Require stack:
- /app/node _modules/@aws-sdk/client-s3/dist-cjs/endpoint/endpointResolver.js
- /app/node_modules/daws-sdk/client-s3/dist-cjs/runtimeConfig.shared.js
/app/node_modules/@aws-sdk/client-s3/dist-cjs/runtimeConfig.js
- /app/node_modules/daws-sdk/client-s3/dist-cjs/S3Client.js
/app/node_modules/daws-sdk/client-s3/dist-cjs/53.js
/app/node_modules/daws-sdk/client-s3/dist-cjs/index.js
/app/build/main/backend/external/aws/aws.service.js
/app/build/main/rest/routers/media-assets.route.js
/app/build/main/rest/rest-api.server.js
/app/build/main/index.js
at Function.Module. resolveFilename (node: internal/modules/cis/loader:956:15)
at Module.require (/app/node modules/dd-trace/packages/dd-trace/src/ritm.js:15:29)
at require (node: internal/modules/cjs/helpers:102:18)
at Object.<anonymous> (/app/node_modules/daws-sdk/client-s3/dist-cjs/endpoint/endpointResolver.js:4:26
at Module._compile (node: internal/modules/cjs/loader:1126:14)
at Object. Module._extensions..js (node: internal/modules/cjs/loader:1180:10)
at Module.load (node: internal/modules/cjs/loader:1004:32)
at Function.Module. load (node: internal/modules/cjs/loader:839:12)
at Module.require (node: internal/modules/cjs/loader:1028:19)
at Module.Hook. require.Module.require (/usr/local/lib/node modules/pm/node modules/require-in-the-middle/index.js:101:39){
code: 'MODULE NOT FOUND'
requireStack: [
"/app/node modules/@aws-sdk/client-s3/dist-cjs/endpoint/endpointResolver.js
```

**Comment by sali141** [comment]
> I have the same problem with amazon-chime-sdk-js@3.8.0 when adding @aws-sdk/util-endpoints@3.193.0 to try to fix the issue: #4060
> 
> `Attempted import error: 'partitions'.'find' is not exported from './partitions.json' (imported as 'partitions').`

I have the same problem.. any updates?

**Comment by yenfryherrerafeliz** [comment]
Hi @usmanghani5995, thanks for opening this issue. Could you all please update to the latest version. I just tested it and the problem is not there, however just let me know otherwise.

Thanks!

**Comment by yenfryherrerafeliz** [comment]
You could either run ```npm update``` or explicitly install the version 3.193.0.

Thanks!

**Comment by nissane** [comment]
I've also tried that.  doesn't seem to fix it.
my package.json has:
    "@aws-sdk/util-endpoints": "^3.193.0",
    "amazon-chime-sdk-component-library-react": "^3.3.0",
    "amazon-chime-sdk-js": "^3.8.0",

doing a npm run build returns:
Attempted import error: 'partitions'.'find' is not exported from './partitions.json' (imported as 'partitions').

**Comment by trivikr** [comment]
AWS SDK for JavaScript maintainer here.

If you're getting `Cannot find module '@aws-sdk/util-endpoints'` error, it's fixed in `v3.193.0`
The verification code is posted in:
* https://github.com/aws/aws-sdk-js-v3/issues/4053#issuecomment-1286137018
* https://github.com/aws/aws-sdk-js-v3/issues/4060#issuecomment-1286136274

For the new error, I'm trying to find a minimal repro. The errors mentioned in this issue are:
* `export 'partitions'.'find' (imported as 'partitions') was not found in './partitions.json' (possible exports: 0, 1, 2, 3, 4)`
* `Attempted import error: 'partitions'.'find' is not exported from './partitions.json' (imported as 'partitions').`
* `Should not import the named export 'partitions' (imported as 'partitions') from default-exporting module (only default export is available soon)`

If you're facing any of this issues, you can unblock yourself by using `v3.190.0` of the SDK.
If you have a minimal repro available, please post it as a comment in this issue. I suspect, the issue if with bundler.

**Comment by trivikr** [comment]
We have a fix attempt out in https://github.com/aws/aws-sdk-js-v3/pull/4072, and we're pretty confident it would fix this issue based on existing default imports of JSON files. The fix will be available with `v3.194.0` within couple of hours.

If you have a minimal repro, please post it as a comment in this issue so that we can use it for verification.

**Comment by yenfryherrerafeliz** [comment]
Steps to reproduce:
- Create a react app:
    - ```npx create-react-app repro-app ```
- Install a package with the version ^3.192.0
    - ```npm install @aws-sdk/client-s3```
    - if using yarn then, ```yarn add @aws-sdk/client-s3```
- Import the dependency somewhere in your code:
    - ```import {S3Client} from "@aws-sdk/client-s3"```
- Try to build:
    -  ```npm run build```
    
Error we got:

```console
Creating an optimized production build...
Failed to compile.

Attempted import error: 'partitions'.'find' is not exported from './partitions.json' (imported as 'partitions').
```

**Comment by alexsmith716** [comment]
i made a minimal example here: https://github.com/alexsmith716/minimal-test-example

I tried to deploy to Vercel but failed with message:
```console
2022-10-21T20:04:39.008Z  info  - Linting and checking validity of types...
2022-10-21T20:04:42.762Z  info  - Creating an optimized production build...
2022-10-21T20:04:58.269Z  warn  - Compiled with warnings
2022-10-21T20:04:58.269Z  
2022-10-21T20:04:58.269Z  ./node_modules/@aws-sdk/util-endpoints/dist-es/lib/aws/partition.js
2022-10-21T20:04:58.270Z  Attempted import error: 'partitions'.'find' is not exported from './partitions.json' (imported as 'partitions').
2022-10-21T20:04:58.270Z  
2022-10-21T20:04:58.270Z  Import trace for requested module:
2022-10-21T20:04:58.270Z  ./node_modules/@aws-sdk/util-endpoints/dist-es/lib/aws/partition.js
2022-10-21T20:04:58.270Z  ./node_modules/@aws-sdk/util-endpoints/dist-es/index.js
2022-10-21T20:04:58.271Z  ./node_modules/@aws-sdk/client-s3/dist-es/endpoint/endpointResolver.js
2022-10-21T20:04:58.271Z  ./node_modules/@aws-sdk/client-s3/dist-es/runtimeConfig.shared.js
2022-10-21T20:04:58.271Z  ./node_modules/@aws-sdk/client-s3/dist-es/runtimeConfig.browser.js
2022-10-21T20:04:58.271Z  ./node_modules/@aws-sdk/client-s3/dist-es/S3Client.js
2022-10-21T20:04:58.272Z  ./node_modules/@aws-sdk/client-s3/dist-es/index.js
2022-10-21T20:04:58.272Z  
2022-10-21T20:04:58.272Z  ./node_modules/@aws-sdk/util-endpoints/dist-es/lib/aws/partition.js
2022-10-21T20:04:58.272Z  Should not import the named export 'partitions' (imported as 'partitions') from default-exporting module (only default export is available soon)
2022-10-21T20:04:58.272Z  
2022-10-21T20:04:58.273Z  Import trace for requested module:
2022-10-21T20:04:58.273Z  ./node_modules/@aws-sdk/util-endpoints/dist-es/lib/aws/partition.js
2022-10-21T20:04:58.273Z  ./node_modules/@aws-sdk/util-endpoints/dist-es/index.js
2022-10-21T20:04:58.273Z  ./node_modules/@aws-sdk/client-s3/dist-es/endpoint/endpointResolver.js
2022-10-21T20:04:58.273Z  ./node_modules/@aws-sdk/client-s3/dist-es/runtimeConfig.shared.js
2022-10-21T20:04:58.273Z  ./node_modules/@aws-sdk/client-s3/dist-es/runtimeConfig.browser.js
2022-10-21T20:04:58.274Z  ./node_modules/@aws-sdk/client-s3/dist-es/S3Client.js
2022-10-21T20:04:58.274Z  ./node_modules/@aws-sdk/client-s3/dist-es/index.js
2022-10-21T20:04:58.274Z  
2022-10-21T20:04:58.274Z  ./node_modules/@aws-sdk/util-endpoints/dist-es/lib/aws/partition.js
2022-10-21T20:04:58.274Z  Should not import the named export 'partitions' (imported as 'partitions') from default-exporting module (only default export is available soon)
2022-10-21T20:04:58.274Z  
2022-10-21T20:04:58.274Z  Import trace for requested module:
2022-10-21T20:04:58.275Z  ./node_modules/@aws-sdk/util-endpoints/dist-es/lib/aws/partition.js
2022-10-21T20:04:58.275Z  ./node_modules/@aws-sdk/util-endpoints/dist-es/index.js
2022-10-21T20:04:58.275Z  ./node_modules/@aws-sdk/client-s3/dist-es/endpoint/endpointResolver.js
2022-10-21T20:04:58.275Z  ./node_modules/@aws-sdk/client-s3/dist-es/runtimeConfig.shared.js
2022-10-21T20:04:58.275Z  ./node_modules/@aws-sdk/client-s3/dist-es/runtimeConfig.browser.js
2022-10-21T20:04:58.275Z  ./node_modules/@aws-sdk/client-s3/dist-es/S3Client.js
2022-10-21T20:04:58.276Z  ./node_modules/@aws-sdk/client-s3/dist-es/index.js
2022-10-21T20:04:58.276Z  
2022-10-21T20:04:58.276Z  info  - Collecting page data...
2022-10-21T20:05:02.072Z  
2022-10-21T20:05:02.072Z  > Build optimization failed: found page without a React Component as default export in 
2022-10-21T20:05:02.073Z  pages/fetchMe
2022-10-21T20:05:02.073Z  
2022-10-21T20:05:02.073Z  See https://nextjs.org/docs/messages/page-without-valid-component for more info.
2022-10-21T20:05:02.073Z  
2022-10-21T20:05:02.116Z  error Command failed with exit code 1.
2022-10-21T20:05:02.116Z  info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
2022-10-21T20:05:02.140Z  Error: Command "yarn run build" exited with 1
```

**Comment by trivikr** [comment]
Verified that the issue is fixed in `v3.194.0`

```console
$ workspace> npx create-react-app repro-app 
Need to install the following packages:
  create-react-app@5.0.1
Ok to proceed? (y) y
...
We suggest that you begin by typing:

  cd repro-app
  npm start

Happy hacking!

$ workspace> cd repro-app

$ repro-app> npm install @aws-sdk/client-s3@3.193.0 --save-exact

$ repro-app> sed -i '1s/^/import {S3Client} from "@aws-sdk\/client-s3"\n/' src/index.js

$ repro-app> npm run build
...
Creating an optimized production build...
Failed to compile.

Attempted import error: 'partitions'.'find' is not exported from './partitions.json' (imported as 'partitions').

$ repro-app> npm install @aws-sdk/client-s3@3.194.0 --save-exact

$ repro-app> npm run build
...
Creating an optimized production build...
Compiled with warnings.

[eslint] 
src/index.js
  Line 1:9:  'S3Client' is defined but never used  no-unused-vars
...
```

**Comment by trivikr** [comment]
Closing as the issue is resolved, and verified using create-react-app in https://github.com/aws/aws-sdk-js-v3/issues/4067#issuecomment-1287473525

Please create a new [bug report](https://github.com/aws/aws-sdk-js-v3/issues/new/choose) with a minimal repro, if any new issue exists in your bundler setup.
